### PR TITLE
Adds tests and improves docs for the parameter class

### PR DESCRIPTION
**Pull request by StephenNneji**
Adds tests for the parameters class
Refactors addParam to remove multiple type checks
Fixes typos

### CODE REVIEW

File: API/projectClass/parametersClass.m [hunk 1]
```diff
@@ -54,94 +71,79 @@
...
+                    error('Expecting numeric values as params 2 - 4');
                 end
+                
+                if ~islogical(fit)
+                    error('Expecting logical value for param 5')
+                end
+                
+                if ~strcmpi(priorType,{'uniform', 'gaussian', 'jeffreys'})
+                    error('Prior type must be ''uniform'', ''gaussian'' or ''jeffreys''');
+                end
```
Comment (DrPaulSharp) [inline API/projectClass/parametersClass.m:131]: So this string parameter can be passed as a string without any problems, unlike the parameter name.

File: tests/testParametersClass.m [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            params = parametersClass(testCase.parameters(1, :));
+            params.paramsTable = [params.paramsTable; vertcat(testCase.parameters(2:end, :))];
+            paramNames = convertCharsToStrings(testCase.parameters(:, 1));
+            % Checks that parameter can be removed
+            testCase.verifyError(@() params.removeParam(), ?MException);
+            testCase.verifyError(@() params.removeParam('bad name'), ?MException);
+            params.removeParam(1);
+            testCase.verifySize(params.paramsTable, [8, 8], 'Parameters has wrong dimension');
+            testCase.verifyEqual(params.paramsTable{:, 1}, paramNames(2:end, 1), 'removeParam method not working');
+            params.removeParam(8, 7, 6);  % is this intended?
```
Comment (DrPaulSharp) [inline tests/testParametersClass.m:102]: Might be worth testing if a sequence in ascending order works - does it crash due to indices going beyond the maximum?

**Review by DrPaulSharp — APPROVED** [review]

File: tests/testParametersClass.m [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+            testCase.verifyTrue(params.paramsTable{1, 5}, 'setParameter method not working');   
+            testCase.verifyError(@() params.setFit({'Not present', false}), ?MException)
+            % Checks that parameter priors can be modified
+            testCase.verifyError(@() params.setPrior({1, '2'}), ?MException);
+            params.setPrior({1, 'gaussian', 1, 2});
+            testCase.verifyEqual(params.paramsTable{1, 6:8}, ["gaussian", 1, 2], 'setParameter method not working');
+            params.setPrior({'Heads', 'uniform'});
+            testCase.verifyEqual(params.paramsTable{1, 6:8}, ["uniform", 0, Inf], 'setParameter method not working');
+            params.setPrior({'Heads', 'jeffreys'});
+            % testCase.verifyEqual(params.paramsTable{1, 6}, "jeffries", 'setParameter method not working');
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply tests/testParametersClass.m:155]: Yes, I added it to our issues doc then forgot to mention it in the meeting

File: API/projectClass/parametersClass.m [hunk 4]
```diff
@@ -54,94 +71,79 @@
                 
                 % First input must be a parameter name
                 if ~ischar(inputCell{1})
-                    error('First value must be param name');
+                    error('First value must be param name (char)');
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply API/projectClass/parametersClass.m:74]: To the best of my knowledge, the MATLAB `table` does not support char array so it has to be a string. We might have to do a refactor so everything is a string and char arrays are casted to strings

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply API/projectClass/parametersClass.m:74]: Yeah, I looked into this, MATLAB does warn against using char in tables. I'm just not sure why a string should be disallowed as an input at this point given that.

File: API/projectClass/parametersClass.m [hunk 5]
```diff
@@ -153,19 +155,21 @@
...
             inputValues = varargin{:};
             tab = obj.paramsTable;
             
             if ischar(inputValues{1})
                 name = string(inputValues{1});
-                row = findRowIndex(name,tab);
+                row = obj.findRowIndex(name,tab);
             else
                 row = inputValues{1};
             end
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply API/projectClass/parametersClass.m:171]: I will refactor all the instance of name/index checks into its own function and add range checks 
```
            if ischar(inputValues{1})
                name = string(inputValues{1});
                row = obj.findRowIndex(name,tab);
            else
                row = inputValues{1};
            end
```

File: tests/testParametersClass.m (hunk 2 above)
Comment (StephenNneji) [reply tests/testParametersClass.m:102]: Nice catch, I will fix

**Pull request merged**